lib/pure/concurrency/atomics.nim

```diff
@@ -288,10 +288,10 @@ else:
     type
       # Atomic* {.importcpp: "_Atomic('0)".} [T] = object
 
-      AtomicInt8 {.importc: "_Atomic NI8", size: 1.} = object
-      AtomicInt16 {.importc: "_Atomic NI16", size: 2.} = object
-      AtomicInt32 {.importc: "_Atomic NI32", size: 4.} = object
-      AtomicInt64 {.importc: "_Atomic NI64", size: 8.} = object
+      AtomicInt8 {.importc: "_Atomic NI8".} = int8
+      AtomicInt16 {.importc: "_Atomic NI16".} = int16
+      AtomicInt32 {.importc: "_Atomic NI32".} = int32
+      AtomicInt64 {.importc: "_Atomic NI64".} = int64
```

Comment:
This form of wrapping keeps making problems for the Nim compiler. We should try to stick with `object` and find a different solution.

Reply:
> This form of wrapping keeps making problems for the Nim compiler. 

Can you tell me what problems it will cause or where is documented? What if I use distinct int64, will it make a difference?

Reply:
Huge difference yes, `distinct` is fine. The problem is that type **aliases** are not supposed to change the meaning and can be substituted which is not true for an "importc alias" which is a contradiction.

Reply:
I see, thanks!